src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -316,11 +325,18 @@ public function updateAction(Request $request)
                 return $this->redirectHandler->redirectToResource($configuration, $resource);
             }
 
-            if ($configuration->hasStateMachine()) {
-                $this->stateMachine->apply($configuration, $resource);
+            try {
+                $this->resourceUpdater->applyTransitionAndFlush($resource, $configuration, $this->manager);
+            } catch (RaceConditionException $exception) {
```

Comment:
If it's generic behavior and we add another extension layer, how about introduction of a generic exception e.g. `ResourceException` (`RaceConditionException` could extend it) and catch it here? We can have a generic message "Ups, something went wrong, please try again" and it will be possible to reuse this logic for some different use cases than only a race conditions. WDYT?